src/table/headerRenderers/dataHeader.component.tsx

```diff
@@ -91,84 +91,99 @@ const DataHeader = (props: DataHeaderProps): React.ReactElement => {
     </Typography>
   );
 
-  return (
+  const TableCellContent = (provided?: any): React.ReactElement => {
+    return (
+      <TableCell
+        {...provided?.draggableProps}
+        ref={provided?.innerRef}
+        {...provided?.dragHandleProps}
+        size="small"
+        component="th"
+        role="columnheader"
+        sx={sx}
+        variant="head"
+        sortDirection={currSortDirection}
+      >
+        <div
+          style={{
+            overflow: 'hidden',
+            flex: 1,
+          }}
+        >
+          <Box
+            aria-label={`${dataKey} header`}
+            display="flex"
+            onMouseDown={(event) => {
+              // Middle mouse button can also fire onClose
+              if (dataKey.toUpperCase() !== 'ID' && event.button === 1) {
+                event.preventDefault();
+                onClose(dataKey);
+              }
+            }}
+          >
+            {Icon && <Box marginRight={1}>{<Icon />}</Box>}
+            <Box>{inner}</Box>
+          </Box>
+        </div>
+        <div
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            // 40 - enough space for both close icon + divider and some space between them
+            width: '40px',
+            justifyContent: 'space-between',
+          }}
+          onMouseOver={() => {
+            setPermitDragging(false);
+          }}
+          onMouseOut={() => {
+            setPermitDragging(true);
+          }}
+        >
+          {/* If this is an ID header, remove icon visibility but still render it
+          This ensures header widths remain consistent */}
```

Comment:
The problem with this is that the ID column seemingly for no reason covers up the column name when resized very small. I've got a fix for this by using `display: none` and adjusting the `40px` value above

Reply:
That seems to work better, thanks for that